app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,50 @@ def show
     @post = Post.find(params[:id_post])
     @comments = Comment.where(post_id: @post.id)
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+    respond_to do |format|
+      format.html { render :new, locals: { post: @post } }
+    end
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author = current_user
+    if @post.save
+      flash[:success] = 'Post saved successfully'
+      redirect_to "/users/#{current_user.id}/posts/#{@post.id}" # post_path(@post.id)
+    else
+      flash.now[:error] = 'Error: Post could not be saved'
+      render :new
+    end
+  end
+
+  def add_comment
+    @user = current_user
+    @post = Post.find(params[:id_post])
+    comment = Comment.new(params.require(:comment).permit(:text))
+    comment.author = @user
+    comment.post = @post
+    if comment.save
+      flash[:success] = 'Post saved successfully'
+    else
+      flash.now[:error] = 'Error: Post could not be saved'
+    end
+    redirect_to "/users/#{@user.id}/posts/#{@post.id}"
+  end
```

Comment:
- [ ] As required for the project and for the best practices, kindly make sure you have a `CommentsController` file with the `New` and `Create` actions. That means the above code should be implemented in a ``comments_controller`` file. Example:
```ruby
# controllers/comments_controller.rb

class CommentsController < ApplicationController
  ...

  def new
    @comment = Comment.new
  end

 def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    ...
  end


  private 

  def comment_params
    params.require(:comment).permit(:text)
  end

	...
end
```

- [ ] Once you have fixed the above issue, kindly make sure you remembere to add a form in the `views/comments/new.html.erb` file which accepts input for creating comments for the project requirements.